### PR TITLE
Upgrade all Python dependencies

### PR DESCRIPTION
**Pull request by nwalters512**
This was done using this handy command from @mwest1066:

```sh
pip3 install pur
python3 -m pur -r <path-to-requirements.txt> --interactive
```

I'm trying to get a head start on dependabot next week.

### CODE REVIEW

**Review by mwest1066 — APPROVED** [review]

File: images/plbase/python-requirements.txt [hunk 1]
```diff
@@ -11,14 +11,14 @@ isort==5.12.0
...
 matplotlib==3.8.0
-networkx==3.1
+networkx==3.2
 nltk==3.8.1
-numpy==1.26.0
+numpy==1.26.1
 openpyxl==3.1.2
 pandas-stubs==2.1.1.230928
-pandas==2.1.1
+pandas==2.1.2
```
Comment (eliotwrobson) [inline images/plbase/python-requirements.txt:19]: Shouldn't do this yet, should be kept in sync with pandas stubs.

**Review by eliotwrobson — APPROVED** [review]
One minor item but otherwise looks good!

File: images/plbase/python-requirements.txt (hunk 1 above)
Comment (nwalters512) [reply images/plbase/python-requirements.txt:19]: Gotcha, will revert this.

**Review by jonatanschroeder — APPROVED** [review]

File: images/plbase/python-requirements.txt (hunk 1 above)
Comment (nwalters512) [reply images/plbase/python-requirements.txt]: Reverted in a50a696.

**Pull request merged**